src/dbus/freedesktop_screensaver.rs

```diff
@@ -44,15 +45,16 @@ impl ScreenSaver {
         let mut cookie = None;
         for _ in 0..3 {
             // Start from 1 because some clients don't like 0.
-            self.counter = self.counter.wrapping_add(1);
-            if self.counter == 0 {
-                self.counter += 1;
+            self.counter.fetch_add(1, Ordering::SeqCst);
+            if self.counter.load(Ordering::SeqCst) == 0 {
+                self.counter.fetch_add(1, Ordering::SeqCst);
             }
 
-            if let Entry::Vacant(entry) = inhibitors.entry(self.counter) {
+            let inhibitor_key = self.counter.load(Ordering::SeqCst);
```

Comment:
You shouldn't load atomics separately because another thread may be modifying them at the same time. This is why `fetch_add()` both adds and returns the previous value.

Reply:
The way it works is when you call `.fetch_add()`, you get the value that you will use, and the variable is also atomically incremented (so the next time someone calls it, they get their next value).

Reply:
Pushed the change, here is the [code snippet](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=8ef579da8ca2f5a28efeaca73fd9031c) that I use to validate my code, beside testing with the local build.

Reply:
This is what the code should be, please use this:

```rust
            let mut inhibitor_key = self.counter.fetch_add(1, Ordering::SeqCst);
            if inhibitor_key == 0 {
                // Clients don't like 0, add one more.
                inhibitor_key = self.counter.fetch_add(1, Ordering::SeqCst);
            }

            // Remove this line below:
            // let inhibitor_key = self.counter.load(Ordering::SeqCst);
```